hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieHiveUtils.java

```diff
@@ -62,6 +67,7 @@
   public static final String HOODIE_STOP_AT_COMPACTION_PATTERN = "hoodie.%s.ro.stop.at.compaction";
   public static final String INCREMENTAL_SCAN_MODE = "INCREMENTAL";
   public static final String SNAPSHOT_SCAN_MODE = "SNAPSHOT";
+  public static final String VALIDATE_SCAN_MODE = "VALIDATE"; //used for pre-commit validation
```

Comment:
@satishkotha On thinking about this a little deeper, I feel one should be able to do "validate" in both modes `SNAPSHOT` & `INCREMENTAL`. Essentially, what you want to do is a `SNAPSHOT @ commitTime` which is what time travel allows but ensures that we read only committed data. To keep concepts this way, you may want to just have a flag saying `hoodie.%s.consume.uncommitted` whose default value is false, you always fall back to the `HoodieTableFileSystem` with current behavior, if it's set to true, then you do what you are currently doing in "VALIDATE" scan mode. What do you think ?

Reply:
@n3nash Ok, switched config to make it work for all modes in future. I only implemented snapshot mode for now. Incremental part can be added later on after we get feedback from snapshot pre-commit validation and when we have more incremental use cases.